src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperOutput.cs

```diff
@@ -63,31 +61,20 @@ public string TagName
         /// The HTML element's pre content.
         /// </summary>
         /// <remarks>Value is prepended to the <see cref="ITagHelper"/>'s final output.</remarks>
-        public string PreContent { get; set; }
+        public IBufferedStringCollection PreContent { get; }
```

Comment:
suggest creating an `IBufferedTextWriter : ITextWriter` interface and exposing that instead